### PR TITLE
(SIMP-7279) Add EL8 support

### PR DESCRIPTION
**Pull request by trevor-vaughan**
- Updated README.md
- Added REFERENCE.md
- Made use of rsync optional (enabled by default for backwards compatibility)
- Add support for passing in a full dhcpd.conf entry
- Ensure that the SELinux user and type are set for the configuration files
- Switch to using iptables::listen::udp for firewalld compatibility
- Add acceptance test

SIMP-7279 #close
SIMP-5871 #close
SIMP-1356 #close
SIMP-7202 #comment Add EL8 support to simp-dhcp

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
Acceptance test requires https://github.com/simp/pupmod-simp-iptables/pull/72 to run

**Comment by lnemsick-simp** [comment]
Should there be module data for CentOS/RHEL 8?

File: spec/acceptance/nodesets/oel.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - server-el6
+    platform: el-6-x86_64
+    box: onyxpoint/oel-6-x86_64
+    hypervisor: <%= hypervisor %>
+    yum_repos:
+      epel:
+        mirrorlist: 'https://mirrors.fedoraproject.org/metalink?repo=epel-6&arch=$basearch'
+        gpgkeys:
+          - https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-$releasever
+
```
Comment (lnemsick-simp) [inline spec/acceptance/nodesets/oel.yml]: No OEL 8 box yet?

Comment (trevor-vaughan) [reply spec/acceptance/nodesets/oel.yml]: @lnemsick-simp There's a `generic` one but not an `onyxpoint` one so I'm not going down that road yet.

Comment (trevor-vaughan) [reply spec/acceptance/nodesets/oel.yml]: Well, I'll give it a shot and see what happens anyway.

Comment (trevor-vaughan) [reply spec/acceptance/nodesets/oel.yml]: Appeared to work! PR updated.

**Pull request merged**